packages/babel-parser/test/helpers/run-fixture-tests.js

```diff
@@ -0,0 +1,201 @@
+  const difference = new Difference(
+    adjust,
+    onlyCompareErrors ? toJustErrors(expected) : expected,
+    onlyCompareErrors ? toJustErrors(actual) : actual,
+  );
+
+  // No differences means we passed and there's nothing left to do.
+  if (difference === Difference.None) return;
+
+  const error = FixtureError.fromDifference(difference, actual);
+
+  // If we're not overwriting the current values with whatever we get this time
+  // around, then we have a legitimate error that we need to report.
+  if (CI || !OVERWRITE) throw error;
+
+  // We only write the output of the original test, not all it's auto-generated
+  // variations.
+  if (!test.original) return;
+
+  const testLocation = test.taskDir;
+
+  // FIXME: We're just maintaining the legacy behavior of storing *just* the
+  // error `message` here, which differs from the error's `toString()` that we
+  // store for each error in the `errors` array. In both cases, we should
+  // serialize the full error to be able to property test locations,
+  // reasonCodes, etc.
+  const throws = !!actual.threw && actual.threw.message;
+  const optionsLocation = join(testLocation, "options.json");
+
+  // We want to throw away the contents of `throws` here.
+  // eslint-disable-next-line no-unused-vars
+  const { throws: _, ...oldOptions } = readJSON(optionsLocation);
+  const newOptions = { ...oldOptions, ...(throws && { throws }) };
+
+  // Store (or overwrite) the options file if there's anything to record,
+  // otherwise remove it.
+  if (Object.keys(newOptions).length <= 0) {
+    rmf(optionsLocation);
+  } else if (throws) {
+    // The idea here is that we shouldn't need to change anything if this doesn't
+    // throw, and stringify will produce different output than what prettier
+    // wants.
+    writeFileWithNewline(optionsLocation, stringify(newOptions, null, 2));
+  }
+
+  // When only comparing errors, we don't want to overwrite the AST JSON because
+  // it belongs to a different test.
+  if (onlyCompareErrors) return;
+
+  const normalLocation = join(testLocation, "output.json");
+  const extendedLocation = join(testLocation, "output.extended.json");
+
+  const [extended, serialized] = actual.ast ? serialize(actual.ast) : [];
+  const outputLocation =
+    serialized && (extended ? extendedLocation : normalLocation);
+
+  if (outputLocation) {
+    writeFileWithNewline(outputLocation, serialized);
+  }
+
+  // Remove any previous output files that are no longer valid, either because
+  // extension changed, or because we aren't writing it out at all anymore.
+  for (const location of [normalLocation, extendedLocation]) {
+    if (location !== outputLocation) {
+      rmf(location);
+    }
+  }
+}
+
+function readJSON(filename) {
+  try {
+    return JSONParse(readFileSync(filename, "utf-8"));
+  } catch (error) {
+    return {};
+  }
+}
+
+function rmf(path) {
+  try {
+    unlinkSync(path);
+  } catch (error) {}
```

Comment:
I think we should throw if `error.code` is not `ENOENT`. We can only handle the case when `options.json` is not found, for other cases better just propagate the errors.

Reply:
Done: https://github.com/babel/babel/pull/14201/commits/91f73c769cb69f4b9c6df7553046c4a65267118d (Also makes the error message when you expect an error but don't get an error slightly nicer).